external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/internal/OffsetManager.java

```diff
@@ -53,8 +54,8 @@
     public OffsetManager(TopicPartition tp, long initialFetchOffset) {
         this.tp = tp;
         this.initialFetchOffset = initialFetchOffset;
-        this.committedOffset = initialFetchOffset - 1;
-        LOG.debug("Instantiated {}", this);
+        this.committedOffset = initialFetchOffset;
+        LOG.debug("Instantiated {}", this.toString());
```

Comment:
toString() is redundant

Reply:
Yes, but I get a warning about leaking this from the constructor, which this fixes.

Reply:
This is not a big deal. But can you tell me what kinda warning you are talking about? I never heard of such a thing. When you concatenate a string with an object reference, the toString() method of that object reference is called. I am not sure why both approaches are any different.

Reply:
The warning I get is just "leaking this in constructor", Netbeans has the following description: "Using this as parameter can be dangerous in the contructor because the object is not fully initialized.". IntelliJ and Eclipse have similar warnings I believe. 

It's basically saying not to use `this` as a parameter from the constructor, because the called method may use fields that aren't initialized. I agree that leaking `this` from the constructor is not dangerous here, because sfl4j just calls toString, but I'd still like to get rid of the warning. The reason this is flagged is because `this` is used as a parameter to LOG.debug, and not as part of a string concatenation. The IDE can't tell whether `this` is used unsafely in LOG.debug, so it's flagged as a warning.

Reply:
Thanks for the explanation. I just noticed that `this` is called within its own constructor, and therefore makes sense. However, IntelliJ does not give me this warning.